invenio_rdm_records/resources/resources.py

```diff
@@ -76,10 +76,89 @@ def p(route):
                 p(routes["access-request-settings"]),
                 self.update_access_settings,
             ),
+            route("POST", p(routes["delete-record"]), self.delete_record),
+            route("POST", p(routes["restore-record"]), self.restore_record),
+            route("GET", p(routes["list-all"]), self.search_all),
+            route("POST", p(routes["set-record-quota"]), self.set_record_quota),
+            # TODO: move to users?
+            route("POST", routes["set-user-quota"], self.set_user_quota),
         ]
 
         return url_rules
 
+    @request_headers
+    @request_view_args
+    @request_data
+    def set_record_quota(self):
+        """Read the related review request."""
+        item = self.service.set_quota(
+            g.identity,
+            resource_requestctx.view_args["pid_value"],
+            quota_size=resource_requestctx.data["quota_size"],
+            max_file_size=resource_requestctx.data["max_file_size"],
+            notes=resource_requestctx.data["notes"],
+        )
+
+        return {}, 200
+
+    @request_headers
+    @request_view_args
+    @request_data
+    def set_user_quota(self):
+        """Read the related review request."""
+        item = self.service.set_user_quota(
+            g.identity,
+            id_=resource_requestctx.view_args["pid_value"],
+            quota_size=resource_requestctx.data["quota_size"],
+            max_file_size=resource_requestctx.data["max_file_size"],
+            notes=resource_requestctx.data["notes"],
+        )
+
+        return {}, 200
+
+    @request_extra_args
+    @request_search_args
+    @response_handler(many=True)
+    def search_all(self):
```

Comment:
what is the difference between this and the one in `records/drafts-resources`? Why duplicating the method with a new one `search_all`?

Reply:
do you mean .search method ? (I assume yes because there is no .search_all)
the difference is in the fact that search_all returns all records, including deleted and marked for purge, which we don't want from the normal .search method, therefore the difference is in filters. 
I have a vague memory that we said deleted records should be indexed separately, but it was not implemented for some reason, maybe it didn't make the MVP cut. ping @max-moser @lnielsen for more information about the scope of this feature